src/statsmodels/statsmodel.jl

```diff
@@ -48,8 +48,8 @@ for (modeltype, dfmodeltype) in ((:StatisticalModel, DataFrameStatisticalModel),
                                  (:RegressionModel, DataFrameRegressionModel))
     @eval begin
         function StatsBase.fit{T<:$modeltype}(::Type{T}, f::Formula, df::AbstractDataFrame,
-                                              args...; kwargs...)
-            mf = ModelFrame(f, df)
+                                              args...; contrasts::Dict = Dict(), kwargs...)
```

Comment:
I don't think it's just a REPL problem; I got this error initially when I wrote a test for the two-contrast case there. 

Also, the check happens when the `ModelFrame` is constructed, because it converts all the contrasts to `ContrastsMatrix` instances, which only has methods for subtypes of `AbstractContrasts` and `ContrastsMatrix` objects. I suppose the downside of only checking then is that you don't get information in the error message about the variable that contrast belongs to, but that's something we can address after merging if need be.